server/command.go

```diff
@@ -278,34 +278,40 @@ func (p *Plugin) runPopCommand(args []string, extra *model.CommandArgs) (bool, e
 }
 
 func (p *Plugin) runSettingsCommand(args []string, extra *model.CommandArgs) (bool, error) {
-	if len(args) != 2 {
-		p.postCommandResponse(extra, "invalid number of arguments")
-		return true, errors.New("invalid number of arguments")
+	if len(args) < 1 {
+		return true, errors.New("no setting selected")
 	}
 
 	if args[0] == "summary" {
+		if len(args) < 2 {
+			return true, errors.New("choose whether you want this setting `on` or `off`")
+		}
+		if len(args) > 2 {
+			return true, errors.New("too many arguments")
+		}
 		var responseMessage string
 		var err error
 
 		switch args[1] {
 		case "on":
 			err = p.saveReminderPreference(extra.UserId, true)
+			responseMessage = "You will start receiving daily summaries."
 		case "off":
 			err = p.saveReminderPreference(extra.UserId, false)
+			responseMessage = "You will stop receiving daily summaries."
 		default:
-			responseMessage = `invalid input, allowed values for "settings summary" are [on] or [off]"`
-			p.postCommandResponse(extra, responseMessage)
-			return true, errors.New("invalid argument")
+			responseMessage = "invalid input, allowed values for \"settings summary\" are `on` or `off`"
+			return true, errors.New(responseMessage)
 		}
 
 		if err != nil {
 			responseMessage = "error saving the reminder preference"
-			p.postCommandResponse(extra, responseMessage)
-			return false, err
```

Comment:
Should this error get logged to help with debugging?

Reply:
Since we are passing it as a non user error, it is logged in line 114.

Reply:
Maybe I'm misunderstand. In L309 a new error only containing `responseMessage` is returned. Should L309 instead wrap the error?

Reply:
I see what you mean now. In this case I would log the error here, since the other part will also be shown to the user. What do you think?

Reply:
Sound good :+1: